core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4509,6 +4510,7 @@ public static void _doScript(StaplerRequest req, StaplerResponse rsp, RequestDis
             }
 
             try {
+                LOGGER.log(Level.INFO, "Administrative Groovy script by " + Jenkins.getAuthentication().getName() + " from " + req.getRemoteAddr() + ": (base64) " + Base64.getEncoder().encodeToString(text.getBytes()));
```

Comment:
I do not think putting the full script in base64 is a good idea. It might be VERY long. Maybe just capture hash?
If you want to capture the entire script, IMO it should be on more verbose levels

Reply:
See previous discussion why this isn't feasible for the submitter. Basically, they have next to no control over the Jenkins install, configuration, or image, and only get to receive what gets submitted to logs by default.

Unfortunately I got distracted from my efforts to ask around for whether this is a common situation, so I have no idea how common this is.